src/viur/core/session.py

```diff
@@ -199,20 +161,48 @@ def reset(self) -> None:
 
         self.cookie_key = utils.string.random(42)
         self.static_security_key = utils.string.random(13)
-        self.session.clear()
+        self.clear()
         self.changed = True
 
-    def items(self) -> 'dict_items':
+    def __delitem__(self, key: str) -> None:
         """
-            Returns all items in the current session.
+            Removes a *key* from the session.
+            This key must exist.
         """
-        return self.session.items()
+        del self[key]
+        self.changed = True
 
-    def setdefault(self, key: str, value: t.Any) -> t.Any:
+    def __ior__(self, other: dict) -> t.Self:
+        """
+        Merges the contents of a dict into the session.
         """
-            Set a default value in the current session
+        super().__ior__(other)
+        self.changed = True
+        return self
+
+    def update(self, other: dict) -> None:
         """
-        return self.session.setdefault(key, value)
+        Merges the contents of a dict into the session.
+        """
+        self |= other
+
+    def pop(self, key: str, *args, **kwargs) -> t.Any:
+        """
+        Deletes a specified key from the session.
+
+        Use 'default=YourValue' as a parameter to make sure,
+        that a default Value such as 'None' is returned and u dont run into Keyerrors.
+        """
+        default = kwargs.get("default", None)
+
+        if default is None:
+            ret = super().pop(key)
+        else:
+            ret = super().pop(key, default)
+
+        self.changed = True
+
+        return ret
```

Comment:
In your implementation `default` cannot be provided as positional argument and you allow more argument than necessary.
I would work in this case with an sentinel.


```py
_SENTINEL : t.Final[object] = object()
```


```suggestion
    def pop(self, key: str, default=_SENTINEL) -> t.Any:
        """
        Delete a specified key from the session.
        
        If key is in the session, remove it and return its value, else return default. 
        If default is not given and key is not in the session, a KeyError is raised.
        """
        if key in self or default is _SENTINEL:
            value = super().pop(key)
            self.changed = True
            return value
        return default
```

The `KeyError` will be thrown through the super call